### DISCUSSION TITLE
How would you design this schema ?

### DISCUSSION THREAD

**Discussion by psm-solution-group** [ravendb/ravendb#18031] [Q&A] [answered]
Hello,

I have a app where people from companys add their times for one week one time in a week, so every week they add they working times in an week.

```
Collection user: {
  name: string;
  street: string;
  workers_times: [
    {
       "monday":"asdasdsadasdsadasdsadasdas",
       "tuesday":"asdasdsadasdsadasdsadasdas",
       "wednesday":"asdasdsadasdsadasdsadasdas",
       "thursday":"asdasdsadasdsadasdsadasdas",
       "friday":"asdasdsadasdsadasdsadasdas",
       "saturday":"asdasdsadasdsadasdsadasdas",
       "sunday":"asdasdsadasdsadasdsadasdas",
       "user_id":"asdasdsadasdsadasdsadasdas",
    }
   ....
  ]
}
```

But I have read some article about the performance when a document is too large. So at the beginning it is not a problem but when 50x enters every week his times in the document then it can be large in a few months. So it is recommend to use another collection called "user_workers_times" ?

I would be happy to hear your opinion what would be better. I am coming from SQL and I understand maybe the NoSQL Schema design like SQL but it should be not like SQL

ps: for each workers time object is about **1436kb** for each 

4x 1436kb is 5,7mb

5.7mb * 12 = 68mb (so in 1 year the document is 68mb big. I would get problems in a few years because large document are high costs and I should wary about it.

**Comment by ayende** [comment, marked as answer]
That would _be_ a problem, yes. But why are you putting it all in a single document?
A better option would be to do that on a per payroll period, so you have two - four weeks in each document. That matches much more closely the way a payroll system is already working.

See also this blog post for a discussion on a similar topic: https://ayende.com/blog/96257/document-based-modeling-auctions-bids

**Reply by psm-solution-group** [reply]
I have understanding it thank you but in your scenario your action will be at a time ending, in my scenario it will never be end I get constants data from the same user so I cant make a new collection "User times" and then make an array of it and put the times there, I have to create a new document for every week time.

**Reply by ayende** [reply]
I'm not sure that I'm following. Your scenario is time keeping for a user, right?

In that case, you already have a natural boundary for this, the payroll period.

You'll have a document per user per payroll period, with all the times they worked.

**Reply by psm-solution-group** [reply]
Yes, I create a collection "UserTimes" and he have to write his times for every month in advance. So One document is like this:

```
{
  id: UUID;
  userId: UUID;
  monthTimes: string[];
}
...
```

ps: I redesigned it I dont use day names I put now all dates he selected in the array named "monthTimes" like this ["2024-01-24T04:43:36.601Z", "2024-01-24T04:43:36.601Z"]

thanks for help!
